### PR TITLE
Handle connection closing, heartbeat messages and refactor client/connection code

### PR DESCRIPTION
**Pull request by FelipeRosa**

### CODE REVIEW

**Review by lsunsi — APPROVED** [review]
I don't understand fully yet, mas looks good. Some of the code especially in the inner parts looks very nested, I wonder if there is some way of making the core more readable changing organization in some way.

File: src/client.rs [hunk 1]
```diff
@@ -65,12 +66,13 @@ impl Client {
                 conn_request_tx,
                 conn_subscription_tx,
                 joined_topics: HashSet::new(),
-                reference: 0,
             },
             conn,
         ))
     }
 
+    pub fn close(self) {}
```
Comment (luizParreira) [inline src/client.rs:74]: qq pega com esse close? 🤔

**Review by luizParreira — APPROVED** [review]

File: src/client.rs (hunk 1 above)
Comment (FelipeRosa) [reply src/client.rs:74]: :eyes:

Comment (FelipeRosa) [reply src/client.rs:74]: O `Client` tem vários channels (do tokio) pra comunicar async com a `Connection` (que faz a parte da conexão do websocket e do Phoenix channel). A `Connection` entende que é pra fechar quando o `Client` dropa os senders dos channels que ele fica segurando. Seria só pra ergonomia isso. Pq fechar o client é dropar ele.

**Comment by FelipeRosa** [comment]
> I don't understand fully yet, mas looks good. Some of the code especially in the inner parts looks very nested, I wonder if there is some way of making the core more readable changing organization in some way.

@lsunsi for certeza there is.

**Pull request merged**